### PR TITLE
Use fltk-config to avoid static library

### PR DESCRIPTION
**Pull request by jasp00**
This should fix the problem @midi-pascal has in #2749.

### CODE REVIEW

**Comment by jasp00** [comment]
Builds fixed.

File: plugins/zynaddsubfx/CMakeLists.txt [hunk 1]
```diff
@@ -153,9 +153,15 @@ SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${PLUGIN_DIR}")
 SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 ADD_EXECUTABLE(RemoteZynAddSubFx RemoteZynAddSubFx.cpp "${WINRC}")
 INSTALL(TARGETS RemoteZynAddSubFx RUNTIME DESTINATION "${PLUGIN_DIR}")
-SET(FLTK_FILTERED_LIBRARIES ${FLTK_LIBRARIES})
-LIST(REMOVE_ITEM FLTK_FILTERED_LIBRARIES "${X11_X11_LIB}" "${X11_Xext_LIB}")
-TARGET_LINK_LIBRARIES(RemoteZynAddSubFx zynaddsubfx_gui ZynAddSubFxCore ${FLTK_FILTERED_LIBRARIES} -lpthread )
+
+SET(FLTK_FILTERED_LDFLAGS ${FLTK_LIBRARIES})
+FIND_PROGRAM(FLTK_CONFIG fltk-config)
+IF(FLTK_CONFIG AND NOT (LMMS_BUILD_APPLE OR LMMS_BUILD_WIN32))
```
Comment (tresf) [inline plugins/zynaddsubfx/CMakeLists.txt]: @jasp00 Thanks!

Can we get a small comment in the file about why this is required.... e.g. Remove unnecessary X11 linking or whatever the original issue description/pr description was?

**Comment by zonkmachine** [comment]
> This should fix the problem @midi-pascal has in #2749.

I had the same issue and it looks like #2754 fixes this.

**Comment by tresf** [comment]
I've reviewed this and it looks fine, but first I think we should put a comment in the source code about what the purpose of this is. :+1: Once that's done, please feel free to merge.

**Comment by jasp00** [comment]
Commented.

**Comment by midi-pascal** [comment]
:+1:

**Pull request merged**